reader.lua

```diff
@@ -242,80 +251,77 @@ local function getPathFromURI(str)
     return unescape(str):sub(#prefix+1)
 end
 
--- Get which file to start with
-local last_file = G_reader_settings:readSetting("lastfile")
-local start_with = G_reader_settings:readSetting("start_with")
-local open_last = start_with == "last"
-
-if open_last and last_file and lfs.attributes(last_file, "mode") ~= "file" then
-    UIManager:show(retryLastFile())
-    last_file = nil
-else
-    local QuickStart = require("ui/quickstart")
-    if not QuickStart:isShown() then
-        open_last = true
-        last_file = QuickStart:getQuickStart()
-    end
-end
 
 -- Start app
 local exit_code
 if ARGV[argidx] and ARGV[argidx] ~= "" then
-    local file
     local sanitized_path = getPathFromURI(ARGV[argidx])
     if lfs.attributes(sanitized_path, "mode") == "file" then
         file = sanitized_path
-    elseif open_last and last_file then
-        file = last_file
+    else
+        directory = ARGV[argidx]
```

Comment:
Does the file manager automatically fall back to lfs.currendir() if the path is wrong? It looks a bit odd next to the file one not to have a basic check in place.

Reply:
https://github.com/koreader/koreader/blob/d560f9011d3b4f8bcd2f205019082b6b6c689f1b/frontend/apps/filemanager/filemanager.lua#L1104-L1105

https://github.com/koreader/koreader/blob/d560f9011d3b4f8bcd2f205019082b6b6c689f1b/frontend/apps/filemanager/filemanagerutil.lua#L12-L14

Reply:
Ok, functionally sound, looks slightly odd reading.